bin/demo-server.js

```diff
@@ -0,0 +1,43 @@
+const http = require('http');
+const fs = require('fs');
+const url = require('url');
+
+const processResource = (resource, res) => {
+	fs.createReadStream('build/demo' + resource).pipe(res);
+};
+
+const processModules = (resource, res) => {
+	let params = url.parse(resource, true);
+
+	let modules = params.query.modules;
+
+	modules = modules.replace(/,/g, '_');
+	modules = modules.replace(/\//g, '_');
+
+	let path = 'build/demo/resolutions/' + modules + '.json';
+
+	fs.createReadStream(path).pipe(res);
+};
+
+const server = http.createServer((req, res) => {
+	let resource = req.url;
+
+	if (resource === '/') {
+		resource = '/index.html';
+	}
+
+	if (resource.startsWith('/o/js_resolve_modules')) {
+		processModules(resource, res);
+	} else if (fs.existsSync('build/demo' + resource)) {
```

Comment:
I usually make build and demo scripts sync because it's much easier to implement and read the code. Unless there's a performance justification (f.e: in the build) I don't see value in using async parallelization.

Reply:
I'm fine with leaving this as-is, but I want to point out that it's not about performance (which, like you say, doesn't matter at all in a demo); it's about not leaving an example behind that someone unwittingly copies into a place where performance *does* matter. If you think there's no risk of this being copied, then there's no point in changing it.